63_CUDAInterop/main.cpp

```diff
@@ -214,65 +294,66 @@ class CUDA2VKApp : public examples::MonoDeviceApplication
 			IGPUCommandBuffer::SBufferMemoryBarrier<IGPUCommandBuffer::SOwnershipTransferBarrier> bufBarrier = {
 				.barrier = {
 					.dep = {
-						.dstStageMask = PIPELINE_STAGE_FLAGS::ALL_COMMANDS_BITS,
-						.dstAccessMask = ACCESS_FLAGS::MEMORY_READ_BITS | ACCESS_FLAGS::MEMORY_WRITE_BITS,
+						.dstStageMask = PIPELINE_STAGE_FLAGS::COPY_BIT,
+						.dstAccessMask = ACCESS_FLAGS::TRANSFER_READ_BIT,
 					},
 					.ownershipOp = IGPUCommandBuffer::SOwnershipTransferBarrier::OWNERSHIP_OP::ACQUIRE,
-					.otherQueueFamilyIndex = queue->getFamilyIndex(),
+					.otherQueueFamilyIndex = IQueue::FamilyExternal,
 				},
-				.range = { .buffer = importedbuf, },
+				.range = { .buffer = importedBuf, },
 			};
-
 			bool re = true;
-			re &= cmd->begin(IGPUCommandBuffer::USAGE::ONE_TIME_SUBMIT_BIT);
-			re &= cmd->pipelineBarrier(EDF_NONE, { .bufBarriers = std::span{&bufBarrier,&bufBarrier + 1} });
+			re &= cmd[1]->begin(IGPUCommandBuffer::USAGE::ONE_TIME_SUBMIT_BIT);
+
+			re &= cmd[1]->pipelineBarrier(EDF_NONE, {.bufBarriers = std::span{&bufBarrier,&bufBarrier + 1}});
 
 			IGPUCommandBuffer::SBufferCopy region = { .size = size };
-			re &= cmd->copyBuffer(importedbuf.get(), stagingbuf.get(), 1, &region);
+			re &= cmd[1]->copyBuffer(importedBuf.get(), stagingBufs[0].get(), 1, &region);
 
 			IGPUCommandBuffer::SImageMemoryBarrier<IGPUCommandBuffer::SOwnershipTransferBarrier> imgBarrier = {
 				.barrier = { 
 					.dep = { 
-						.dstStageMask = PIPELINE_STAGE_FLAGS::ALL_COMMANDS_BITS, 
-						.dstAccessMask = ACCESS_FLAGS::MEMORY_READ_BITS | ACCESS_FLAGS::MEMORY_WRITE_BITS,
+						.dstStageMask = PIPELINE_STAGE_FLAGS::COPY_BIT,
+						.dstAccessMask = ACCESS_FLAGS::MEMORY_READ_BITS,
 					},
 					.ownershipOp = IGPUCommandBuffer::SOwnershipTransferBarrier::OWNERSHIP_OP::ACQUIRE,
-					.otherQueueFamilyIndex = queue->getFamilyIndex(),
+					.otherQueueFamilyIndex = IQueue::FamilyExternal,
 				},
-				.image = importedimg.get(),
+				.image = importedImg.get(),
 				.subresourceRange = {
 					.aspectMask = IImage::EAF_COLOR_BIT,
 					.levelCount = 1u,
 					.layerCount = 1u,
 				},
-				.oldLayout = IImage::LAYOUT::UNDEFINED,
+				.oldLayout = IImage::LAYOUT::PREINITIALIZED,
```

Comment:
explain that old layout at the time of release was `PREINITIALIZED` and it needs to match at the time of acquire (external user of an image must put it back to same layout after done according to VK spec)